Allow multiple eos ids

helunwencser opened the pull request:
Summary:
Since llama 3, it could have two eos ids:
```
<end_of_text>
<eot_id>
```
This PR updates the export and runner to allow passing multiple eos ids.

It also fixes `text_token_generator` to not print eos.

Differential Revision: D61420500

larryliu0820 reviewed (changes requested):
Thanks @helunwencser for adding this feature! Since you are on this, can you please add a unit test for `text_token_generator.h`? Also there are a bunch of CI jobs are failing. We used to only depend on CI jobs to guard our code but it seems like unit tests are important for now.

Commit:
Allow multiple eos ids (#4777)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/4777

Since llama 3, it could have two eos ids:
```
<end_of_text>
<eot_id>
```
This PR updates the export and runner to allow passing multiple eos ids.

It also fixes `text_token_generator` to not print eos.

Differential Revision: D61420500
